### PR TITLE
Sync Uri::getScheme() with ServerRequest::scheme().

### PR DESCRIPTION
**Pull request by ADmad**
This ensures that the value returned by `getScheme()` of a `ServerRequest`'s `Uri` instance  honors the proxy trusting config of the request instance.

### CODE REVIEW

**Review by markstory — COMMENTED** [review]
Looks good to me.

File: src/Http/ServerRequestFactory.php [hunk 1]
```diff
@@ -94,6 +90,12 @@ public static function fromGlobals(
         ]);
 
         $request = static::marshalBodyAndRequestMethod($parsedBody ?? $_POST, $request);
+        // This is required as `ServerRequest::scheme()` ignores the value of
```
Comment (swiffer) [inline src/Http/ServerRequestFactory.php:93]: @ADmad I noticed this change broke my setup containing the HttpsEnforcerMiddleware (behind Application Load Balancer).

I now moved HTTPS redirection to the loadbalancer but what steps would be required to ensure HttpsEnforcerMiddlerware trusts Proxy Headers / is it supported at all?

I see that there is a [Section](https://book.cakephp.org/4/en/controllers/request-response.html#trusting-proxy-headers) in the docs but where should modification of the request happen?

Setting [trustProxy in bootstrap.php](https://github.com/cakephp/app/blob/4.x/config/bootstrap.php#L149) is not enough, right?

Comment (ADmad) [reply src/Http/ServerRequestFactory.php:93]: @swiffer The `process()` method of the middleware get's the request instance as argument so use it's `setTrustedProxies()` method there.

Comment (swiffer) [reply src/Http/ServerRequestFactory.php:93]: Got it, would you accept a PR to allow setting an array of trustedProxies as part of the config of httpsenforcermiddleware?

Comment (ADmad) [reply src/Http/ServerRequestFactory.php:93]: Please open a new issue where we can discuss that.

Comment (swiffer) [reply src/Http/ServerRequestFactory.php:93]: #16897

**Pull request merged**